pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -347,7 +348,12 @@ func parseList(legacyList []runtime.Object) (metainternalversion.ListOptions, ma
 }
 
 func enrichLegacyObject(originalObj, returnedObj runtime.Object, created bool) (runtime.Object, error) {
-	accessorReturned, err := meta.Accessor(returnedObj)
+	if returnedObj == nil {
+		return nil, errors.New("cannot enrich nil object")
+	}
+	ret := returnedObj.DeepCopyObject()
```

Comment:
I think this was the root cause at all times: `enrichLegacyObject` takes two objects and returns a third one, but it changes the input at the same time: it is modifying `returnedObj` because it's a pointer type. If `enrichLegacyObject` is expected to actually modify the input, then it should probably not return a `runtime.Object` since the input is already changed, otherwise it could be misleading.

Reply:
yes, we should totally change this fn's signature, I'm making a note to do just that. I learned only too late that these accessors would modify the object. Thanks for fixing that!